inference-engine/ie_bridges/java/sample/face_detection_sample_async/Main.java

```diff
@@ -0,0 +1,241 @@
+        String device = parser.getDevice();
+        int inferRequestsSize = parser.getRequestsNum();
+
+        if(imgsPath == null || xmlPath == null || device == null || inferRequestsSize <= 0) {
+            System.out.println("Error: Wrong argument");
+            return;
+        }
+
+        BlockingQueue<Mat> framesQueue = new LinkedBlockingQueue<Mat>();
+
+        Thread captureThread = new Thread(new Runnable() {
+        
+            @Override
+            public void run() {
+                VideoCapture cam = new VideoCapture();
+                cam.open(imgsPath);
+                Mat frame = new Mat();
+
+                while (true) {
+                    if (cam.read(frame)) {
+                        framesCounter++;
+                        framesQueue.add(frame.clone());           
+                    } else
+                        return;
+                }
+            }
+        });
+
+        Thread inferThread = new Thread(new Runnable() {
+        
+            @Override
+            public void run() {
+                try {
+                    IECore core = new IECore();
+                    CNNNetwork net = core.ReadNetwork(xmlPath);
+    
+                    Map<String, InputInfo> inputsInfo = net.getInputsInfo();
+                    String inputName = new ArrayList<String>(inputsInfo.keySet()).get(0);
+                    InputInfo inputInfo = inputsInfo.get(inputName);
+    
+                    inputInfo.getPreProcess().setResizeAlgorithm(ResizeAlgorithm.RESIZE_BILINEAR);
+                    inputInfo.setLayout(Layout.NHWC);
+                    inputInfo.setPrecision(Precision.U8);
+
+                    String outputName =  new ArrayList<String>(net.getOutputsInfo().keySet()).get(0);
+    
+                    ExecutableNetwork executableNetwork = core.LoadNetwork(net, device);
+                    
+                    Vector<InferRequest> inferRequests = new Vector<InferRequest>();
+                    Vector<Boolean> asyncInferIsFree = new Vector<Boolean>(inferRequestsSize);
+
+                    for (int i = 0; i < inferRequestsSize; i++) {
+                        inferRequests.add(executableNetwork.CreateInferRequest());
+                        asyncInferIsFree.add(true);
+                    }
+
+                    Queue<Integer> startedRequestsIds = new LinkedList<Integer>();
+    
+                    boolean isRunning = true;
+
+                    while (isRunning) {
+                        while (!startedRequestsIds.isEmpty()) {
+                            processInferRequets(outputName, startedRequestsIds, asyncInferIsFree, inferRequests);
+                            break;
+                        }
+
+                        for (int i = 0; i < inferRequestsSize; i++) {
+                            if (!asyncInferIsFree.get(i))
+                                continue;
+
+                            Mat frame = framesQueue.poll(0, TimeUnit.SECONDS);
```

Comment:
This is not reliable. If network faster than frames source we will finish this thread too early. Please use meaningful timeout.

Reply:
Please ignore the statement above. Let's have `while (captureThread.isAlive() || !framesQueue.empty())`. timeout is `0` is OK.